### DISCUSSION TITLE
water electrolyser and fuel cell model

### DISCUSSION THREAD

**Discussion by lucy6678** [pybamm-team/PyBaMM#2464] [Ideas]
Hi,

I am working on modeling water electrolyser for hydrogen generation in python, and come across the PyBaMM package, which I think the framework and solver may also help me.
Before I further look into it, I would like to know, is there any plan to extend pybamm with water electrolyser and fuel cell model?
Or is pybamm framework  suitable for  water electrolyser and fuel cell modeling and simulation, when user try to implement one on their own?
Any further suggestions regarding this topic?

Thank you very much!

**Comment by brosaplanella** [comment]
Hi! We have no plans on extending PyBaMM for electrolyser models, but the existing framework would probably work for such models. Which equations are you trying to implement? You can check the examples about [creating models](https://github.com/pybamm-team/PyBaMM/tree/develop/examples/notebooks/Creating%20Models) which might be helpful if you want to implement the electrolyser models yourself.

**Reply by lucy6678** [reply]
Hi!
thank you very much for the fast reply and help! 
For the equations of interest following paper may give you a better idea
https://iopscience.iop.org/article/10.1149/1945-7111/ac69c4/pdf
such as Stefan-Maxwell equations
The author may have implemented this model in Comsol, so I am thinking if your framework and solver maybe suitable.
And if it is also possible to solve this kind of model further in 3D and transient level with your framework and solver?

**Reply by brosaplanella** [reply]
PyBaMM is designed for simple geometries so it might not be the best option for 3D. For simplified models (like the one in the paper, I guess) it should work fine, but as I said in my previous comment you should implement the model from scratch.

**Reply by lucy6678** [reply]
Thank you very much for your information!
